src/probnum/filtsmooth/gaussfiltsmooth/kalman_utils.py

```diff
@@ -218,3 +306,96 @@ def rts_smooth_step_classic(
         + smoothing_gain @ (smoothed_rv.cov - predicted_rv.cov) @ smoothing_gain.T
     )
     return pnrv.Normal(new_mean, new_cov), {}
+
+
+def rts_smooth_step_sqrt(
+    unsmoothed_rv,
+    predicted_rv,
+    smoothed_rv,
+    crosscov,
+    dynamics_model=None,
+    start=None,
+    stop=None,
+) -> (pnrv.RandomVariable, typing.Dict):
+    r"""Smoothing step in square-root form.
+
+    Assumes a prior dynamic model of the form
+
+        .. math:: x \\mapsto N(A x, Q).
+
+    For the mathematical justification of this step, see Eq. 45 in
+    https://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.68.1059&rep=rep1&type=pdf.
+    """
+    A, SQ, shift = linear_system_matrices(dynamics_model, None, start, stop, None)
+    SC_past = unsmoothed_rv.cov_cholesky
+    SC_futu = smoothed_rv.cov_cholesky
+    G = crosscov @ np.linalg.inv(predicted_rv.cov)
+
+    dim = len(A)
+    zeros = np.zeros((dim, dim))
+    blockmat = np.block(
+        [
+            [SC_past.T @ A.T, SC_past.T],
+            [SQ.T, zeros],
+            [zeros, SC_futu.T @ G.T],
+        ]
+    )
+    big_triu = np.linalg.qr(blockmat, mode="r")
+    SC = big_triu[dim : 2 * dim, dim:]
+
+    new_cov_cholesky = triu_to_positive_tril(SC)
+    new_cov = new_cov_cholesky @ new_cov_cholesky.T
+    new_mean = unsmoothed_rv.mean + G @ (smoothed_rv.mean - predicted_rv.mean)
+    return pnrv.Normal(new_mean, new_cov, cov_cholesky=new_cov_cholesky), {}
+
+
+########################################################################################################################
+# Helper functions
+########################################################################################################################
+
+
+def linear_system_matrices(model, rv, start, stop, _linearise_at):
```

Comment:
The function seems a bit weird to me. At least I'm still a bit unsure about how I see the whole "start/stop" notion in these transitions. In discrete models, we pass `stop=None`, which feels surprising at first sight (do we just transition forever?). But I mean I totally understand why the current implementation also makes sense, and I also don't really see a better alternative.

Reply:
This start stop malarkey is set into stone via #313 now